components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/extensions/AgentResourceManager.java

```diff
@@ -0,0 +1,309 @@
+ * 
+ * <p>Key features:</p>
+ * <ul>
+ *   <li>Agent creation with automatic password generation</li>
+ *   <li>Domain-specific agent management (AGENT domain)</li>
+ *   <li>Extended schema validation for agent-specific attributes</li>
+ *   <li>SCIM 2.0 protocol compliance for agent operations</li>
+ * </ul>
+ * 
+ * @since 7.2.0
+ * @see UserResourceManager
+ * @see org.wso2.charon3.core.protocol.endpoints.AbstractResourceManager
+ */
+public class AgentResourceManager extends UserResourceManager {
+
+    /** Logger instance for this class. */
+    private static final Logger logger = LoggerFactory.getLogger(AgentResourceManager.class);
+     
+    /** REST endpoint path for agent resources. */
+    private static final String AGENT_ENDPOINT = "/Agents";
+    
+    /** Default domain name for agent user store operations. */
+    protected static final String AGENT_STORE_DOMAIN = "AGENT";
+
+    /**
+     * Constructs a new AgentResourceManager instance.
+     * 
+     * <p>Initializes the agent resource manager by calling the parent UserResourceManager
+     * constructor and logging the initialization for debugging purposes.</p>
+     */
+    public AgentResourceManager() {
+        super();
+        logger.debug("AgentResourceManager initialized for agent management operations");
+    }
+
+    /**
+     * Creates a new agent resource with SCIM 2.0 protocol compliance.
+     * 
+     * <p>This method handles the complete agent creation lifecycle including:</p>
+     * <ul>
+     *   <li>SCIM object decoding and validation</li>
+     *   <li>Domain name prefix addition for agent store</li>
+     *   <li>Automatic secure password generation if not provided</li>
+     *   <li>Agent-specific schema validation</li>
+     *   <li>Delegation to external agent manager for persistence</li>
+     *   <li>Response encoding with proper headers</li>
+     * </ul>
+     * 
+     * @param scimObjectString  Raw JSON string containing agent information in SCIM format
+     * @param agentManager      Agent manager instance for handling agent persistence operations
+     * @param attributes        Comma-separated list of attributes to include in response
+     * @param excludeAttributes Comma-separated list of attributes to exclude from response
+     * @return SCIM response containing created agent data with HTTP 201 status
+     * @throws CharonException if SCIM processing fails
+     * @throws BadRequestException if agent data is invalid
+     * @throws ConflictException if agent already exists
+     * @throws InternalErrorException if server processing fails
+     */
+    @Override
+    public SCIMResponse create(String scimObjectString, UserManager agentManager, String attributes,
+            String excludeAttributes) {
+
+        try {
+            logger.debug("Starting agent creation process");
+            
+            // Obtain the JSON encoder for response formatting
```

Comment:
```suggestion
            // Obtain the JSON encoder for response formatting.
```
Check other places as well.